app/renderer.js

```diff
@@ -107,10 +107,27 @@ module.exports = class Renderer {
     // ## Public methods
     // Renders given file with given options
     renderFile(files, options, done) {
+        let promise;
+
         if (!done && typeof options === "function") {
             done = options;
             options = {};
         }
+
+        if (!done && typeof Promise === 'function') {
+            promise = new Promise((resolve, reject) => {
+                done = (err) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve();
+                    }
+                };
+            });
+        }
+
+        done = done || function () {};
```

Comment:
Wouldn't it be easier to call resolve or reject inside the last element of the callback nest instead of changing done?
```js
let promise=new Promise((resolve,reject)=>{
  // some code
  this.fileOutput(res, path.join(renderOptions.outputDirectory, renderOptions.outputFilename), (err,res)=>{
    if(done) done(err,res);
    if(err || !res) reject(err);
    else resolve(res);
  });
});
return promise;
```